src/org/zaproxy/zap/extension/ascanrulesBeta/XXEPlugin.java

```diff
@@ -207,23 +207,17 @@ public int getRisk() {
         return Alert.RISK_HIGH;
     }
 
-    @Override
-    public void init() {
-        if (pluginApi == null) {
-            ExtensionAscanRulesBeta ext = Control.getSingleton()
-                    .getExtensionLoader()
-                    .getExtension(ExtensionAscanRulesBeta.class);
-            pluginApi = ext.getXXEPluginAPI();
-        }
-    }
-
     /**
      * Plugin to scan for XXE vulnerabilities.
      * It checks both for local and remote using the ZAP API
      * and also a new model based on parameter substitution
      */
     @Override
     public void scan() {
+        if (pluginApi.getExtensionCallback() == null) {

```

Comment:
It could do this check in `init()` instead and skip the scanner with a reason (`getParent().pluginSkipped(this, "...");`), it would inform the user of why it didn't do anything.